pkg/controller/statefulset/stateful_set_control_test.go

```diff
@@ -227,6 +228,28 @@ func CreatesPods(t *testing.T, set *apps.StatefulSet, invariants invariantFunc)
 	if set.Status.UpdatedReplicas != 3 {
 		t.Error("Failed to set UpdatedReplicas correctly")
 	}
+	// Check all pods have correct pod index label.
+	if feature.DefaultFeatureGate.Enabled(features.PodIndexLabel) {
```

Comment:
Reviewer note: I thought about adding a `podIndexLabelEnabled bool` as a test case parameter, but that would then require changing the function signature of all 11 test functions (e.g. CreatePods, ScalesUp, ScalesDown, etc.) to add this parameter. This seemed like I would be shoe-horning it in and making the test code uglier, so I opted to just check if the feature gate itself is enabled in the test code, even though I don't see this pattern used much elsewhere in the codebase. 

Let me know if there is ok or if there is a different/preferred pattern to do this.

Reply:
cc @soltysh on above comment